src/cli/crd/create.test.ts

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import { expect, describe, it, jest, beforeEach } from "@jest/globals";
+import { validateScope, generateCRDScaffold } from "./create";
+
+describe("generateCRDScaffold", () => {
+  it("should generate the correct CRD scaffold", () => {
+    const group = "example";
+    const version = "v1alpha1";
+    const kind = "MyCustomResource";
+    const data = {
+      domain: "example.com",
+      plural: "mycustomresources",
+      scope: "Namespaced",
+      shortName: "mcr",
+    };
+
+    const result = generateCRDScaffold(group, version, kind, data);
+    expect(result).toContain(`${kind}Spec`);
+    expect(result).toContain(`${kind}Status`);
+    expect(result).toContain(`// Group: ${group}`);
+    expect(result).toContain(`// Version: ${version}`);
+    expect(result).toContain(`// Domain: ${data.domain}`);
+    expect(result).toContain(`plural: "${data.plural}"`);
+    expect(result).toContain(`scope: "${data.scope}"`);
+    expect(result).toContain(`shortName: "${data.shortName}"`);
+    expect(result).toContain(`${kind}StatusCondition`);
+    expect(result).toContain(`${kind}Status`);
```

Comment:
This is thorough, I like it. What do you think of making these strings a collection that we iterate over? Perhaps something along these lines:

```suggestion
const expectedSnippets = [
  // Kind-related
  `${kind}Spec`,
  `${kind}Status`,
  `${kind}StatusCondition`,
  `${kind}Status`,

  // Metadata comments
  `// Group: ${group}`,
  `// Version: ${version}`,
  `// Domain: ${data.domain}`,

  // CRD config fields
  `plural: "${data.plural}"`,
  `scope: "${data.scope}"`,
  `shortName: "${data.shortName}"`,
];

expect(expectedSnippets.every(snippet => result.includes(snippet))).toBe(true);  
```

The value here is that we show readers of the test what sorts of things we care about in a CRD definition (e.g., kind-related thing, metadata, and CRD config fields)

Reply:
I see where you are thinking, this exact suggestion would not work due to those comments which do not actually exist in the string that the function uses to scaffol (would fail on every) for things like `// CRD config fields`.

This is a good idea so here is the issue for it so we can tackle it in the future since we have limited resources this week.

https://github.com/defenseunicorns/pepr/issues/2053